comp/otelcol/collector/pipeline.go

```diff
@@ -88,7 +88,6 @@ func (c *collector) Start() error {
 	c.col = col
 	// the context passed to this function has a startup deadline which
 	// will shutdown the collector prematurely
-	ctx := context.Background()
```

Comment:
Has this been tested? To my understanding, the `ctx` context passed to `start` by fx is a timeout context, meaning it will be cancelled in a few seconds. This is to ensure that OnStart hooks only run a short amount of time. The reason this code was using the `context.Background` instead is so that `col.Run` could act as a long-running task.

See the fx docs where this is mentioned: https://uber-go.github.io/fx/lifecycle.html#lifecycle-hooks

Reply:
Thanks for catching this, just reverted this change